scripts/detect_artefacts.py

```diff
@@ -0,0 +1,311 @@
+                                  1.5,
+                                  ji[str(int(tg1[ind_2]))][0], 2, )
+            theimagee = Image.fromarray((imm * 255).astype(np.uint8))
+            theimagee.save(f'{save_to_folder}/{val}')
+        if code_KIE:
+            for i in range(len(tg)):
+                txt = []
+                if int(tg1[i]) == 1:
+                    pio = kold[int(tg[i][1]) - 6:int(tg[i][3]) + 6, int(tg[i][0]) - 6:int(tg[i][2]) + 6]
+                    try:
+                        txt = reader.decode_array(pio)
+                        if "parsed" in list(txt[0].keys()):
+                            code_info.append(txt[0]["parsed"].decode())
+                        else:
+                            code_info.append(["Unreadable/Not a code"])
+                    except ZeroDivisionError:
+                        code_info.append(["Unreadable/Not a code"])
+
+                elif int(tg1[i]) == 2:
+                    pio = kold[int(tg[i][1]) + 10:int(tg[i][3]) - 5, int(tg[i][0]) - 2:int(tg[i][2]) + 2]
+                    try:
+                        txt = decode(pio)
+                        code_info.append(txt)
+                    except KeyError:
+                        code_info.append(["Unreadable/Not a code"])
+
+            inf_dic.update(
+                {"geometry": str(bbox), "class": str(art_class), "score": str(c_score), "KIE": str(code_info)})
+        else:
+            inf_dic.update({"geometry": str(bbox), "class": str(art_class), "score": str(c_score)})
+    return inf_dic
+
+
+def main(args):
+    torch.backends.cudnn.benchmark = True
+
+    # Filter keys
+    state_dict = {
+        k: v for k, v in torchvision.models.detection.__dict__[args.arch](pretrained=True).state_dict().items()
+        if not k.startswith('roi_heads.')
+    }
+    defaults = {"min_size": 800, "max_size": 1300,
+                "box_fg_iou_thresh": 0.5,
+                "box_bg_iou_thresh": 0.5,
+                "box_detections_per_img": 150, "box_score_thresh": 0.15, "box_positive_fraction": 0.35,
+                "box_nms_thresh": 0.2,
+                "rpn_pre_nms_top_n_train": 2000, "rpn_pre_nms_top_n_test": 1000,
+                "rpn_post_nms_top_n_train": 2000, "rpn_post_nms_top_n_test": 1000,
+                "rpn_nms_thresh": 0.2,
+                "rpn_batch_size_per_image": 250
+                }
+    kwargs = {**defaults}
+
+    model = torchvision.models.detection.__dict__[args.arch](pretrained=False, num_classes=5, **kwargs)
+    model.load_state_dict(state_dict, strict=False)
+    model.roi_heads.box_roi_pool = MultiScaleRoIAlign(featmap_names=['0', '1', '2', '3'], output_size=(7, 7),
+                                                      sampling_ratio=2)
+    anchor_sizes = ((16), (64), (128), (264))
+    aspect_ratios = ((0.5, 1.0, 2.0, 3.0,)) * len(anchor_sizes)
+    model.rpn.anchor_generator.sizes = anchor_sizes
+    model.rpn.anchor_generator.aspect_ratios = aspect_ratios
```

Comment:
Would you mind using the faster_rcnn already in the repo please?

Reply:
Yes, resolved . Thanks